components/Error.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import Layout from './Layout'
+import NavLink from './NavLink'
+import { Text } from './theme/Text'
+type ErrorProps = {
+  title: string
+  src: string
+  message?: string
+}
+const Error: React.FC<ErrorProps> = ({ title, message, src }) => {
```

Comment:
Does the `message` and image `src` have a relationship with one another? Like, every time message equals `No data`, do we want to display the 500 error image?

Similarly, when the message equals `404 error`, do we want to `always` display the `404` error image? If we have images that are `only` used for specific error messages or titles, then I don't think we should pass in `src` as a prop. Instead, you can have some kind of table where you select the `src` based on the `message` or `title` passed in. 

If it is the case that the image `src` is not bound to any error message or title(`like display 404 error image when message equals 404 error`), then I think it would be better to have `src` have only a limited number of options(`notFound, 404`) as a valid input. And instead of `src` it should be renamed something like `type`. A table can be made to match the corresponding image `src` depending on value of `type` passed in. This would make the component easier for others to use in the future, since they won't have to determine the correct paths to use as a value for `src`.

Ex
```
const imageSources = {
  notFound: 'notFound.png',
  404: '404.png'
}
```

Reply:
Titles and background images are tied together. `Page not found` is shown with a confused man, `Internal server error` results in generic server image, so it makes sense to merge them in a new type somehow. I am not sure if enum is the best way to do so though, why not a simple union?
```
 type ErrorProps = {
  type: '404' | '500'
  message?: string
}
const Error: React.FC<ErrorProps> = ({ type, message }) => {
       ...
              <Text component="div" size="xl" bold={true}>
                {type === '404' ? 'Page not found' : 'Internal server error'}
              </Text
              ....
              ```

Reply:
Yep, unions look like they work too! I apologize for the misunderstanding, I should have been more clear in my language. I only used `enums` as an example, which is why I said `something like this`

Reply:
OK, I changed it to union type. If we need more error pages we could extend this union and change ternary operator for a switch.